### ISSUE TITLE
Actions using wrong URL for Repositories

### ISSUE THREAD

**Issue by OdinVex** [go-gitea/gitea#26721]
### Description

I host my Gitea instance on a subdir of a domain, such as `https://example.com/Gitea/` and I've noticed that Actions isn't giving the correct path to the Runner (or jobs?). Gitea Actions Demo was a perfect test and it revealed that it's passing the wrong URL for the repository:

Explore-Gitea-Actions (demo.yaml) relevant line:
```
remote add origin https://example.com/USER/REPO
```
It should be `https://example.com/Gitea/USER/REPO` in my case. It's broken the ability of git to get the repo into the job.

### Gitea Version

1.20.3

### Can you reproduce the bug on the Gitea demo site?

No

### Git Version

2.39.2

### Operating System

Docker Gitea & Gitea Actions

### How are you running Gitea?

Multiple ways, but primary test instance is self-hosted via Docker on a reverse-proxy.

### Database

None

**Comment by sosyco** [comment]
similar configuration, same problem.

**Comment by natesubra** [comment]
This bit me as well. 

I have Gitea behind a reverse proxy (Traefik) using a suburl/prefix. Clone addresses etc were fine--and I have the base URL set in my app.ini. Mainly commenting to help make this more discoverable.

Swapping over to one of the forked checkouts does indeed work. Instructions can be found on the README.md of both alternative projects.

**Comment by natesubra** [comment]
> Microsoft's upstream GitHub action `checkout` "developers" refuse to adjust the checkout to address this issue. They're essentially telling everyone to **** off, so it'll need forking to un**** it.

Yeah it's definitely frustrating-- fingers crossed it will make it's way in eventually. Sucks to have to shim an action in nearly every workflow.

**Comment by wxiaoguang** [comment]
There are 2 assumptions in GitHub actions ecosystem:

1. there is no sub-path
2. the repo path must be `owenr/repo` (no sub-repo)

So it seems no way to change the situation.
